partitura/utils/music.py

```diff
@@ -172,82 +172,58 @@ def _transpose_note_inplace(note, interval):
-        if interval.direction == "up":
-            diff_sm = tmp_pc - prev_pc if tmp_pc >= prev_pc else tmp_pc + 12 - prev_pc
-        else:
-            diff_sm = prev_pc - tmp_pc if prev_pc >= tmp_pc else prev_pc + 12 - tmp_pc
-        note.alter = (
-            INTERVAL_TO_SEMITONES[interval.quality + str(interval.number)] - diff_sm
+        new_step, new_alter, new_octave = transpose_note_attributes(
+            interval=interval,
+            step=note.step,
+            alter=note.alter,
+            octave=note.octave,
         )
-
-
-def transpose_note_old(step, alter, interval):
-    """
-    Transpose a note by a given interval without changing the octave or creating a Note Object.
-
-
-    Parameters
-    ----------
-    step: str
-        The step of the pitch, e.g. C, D, E, etc.
-    alter: int
-        The alteration of the pitch, e.g. -2, -1, 0, 1, 2 etc.
-    interval: Interval
-        The interval to transpose by.
-
-    Returns
-    -------
-    new_step: str
-        The new step of the pitch, e.g. C, D, E, etc.
-    new_alter: int
-        The new alteration of the pitch, e.g. -2, -1, 0, 1, 2 etc.
-    """
-    if interval.quality + str(interval.number) == "P1":
-        new_step = step
-        new_alter = alter
-    else:
-        prev_step = step.capitalize()
-        new_step = _transpose_step(prev_step, interval.number, interval.direction)
-        prev_alter = alter if alter is not None else 0
-        prev_pc = MIDI_BASE_CLASS[prev_step.lower()] + prev_alter
-        tmp_pc = MIDI_BASE_CLASS[new_step.lower()]
-        if interval.direction == "up":
-            diff_sm = tmp_pc - prev_pc if tmp_pc >= prev_pc else tmp_pc + 12 - prev_pc
-        else:
-            diff_sm = prev_pc - tmp_pc if prev_pc >= tmp_pc else prev_pc + 12 - tmp_pc
-        new_alter = (
-            INTERVAL_TO_SEMITONES[interval.quality + str(interval.number)] - diff_sm
+        note.step = new_step
+        note.alter = new_alter
+        note.octave = new_octave
+        if update_ties:
+            for tied_note in note.tie_next_notes:
+                _transpose_note_inplace(tied_note, interval, update_ties=False)
+
+
+def _transpose_ks_inplace(ks: KeySignature, interval: Interval):
+    key = ks.name
+    step = key[0]
+    alter = key.count("#") - key.count("b")
+    new_step, new_alter, _ = transpose_note_attributes(interval, step=step, alter=alter)
+    new_fifths, mode = key_name_to_fifths_mode(new_step + ALTER_SIGNS[new_alter])
+    if abs(new_fifths) > 7:
+        raise ValueError(
+            f"Key signature {ks.name} transposed by {interval} results in an invalid "
+            f"key signature with {new_fifths} fifths. The maximum number of fifths is "
+            f"+/- 7.",
         )
-    return new_step, new_alter
+    ks.fifths = new_fifths
 
 
-def transpose_note(step, alter, interval):
+def transpose_note_attributes(
+    interval: Interval,
+    step: str,
+    alter: int = 0,
+    octave: int = 0,
+):
```

Comment:
I recommend not to deprecate the `transpose_note` function. You can add a warning or future deprecation flag as a decorator on top of the function `@deprecated("Starting with version X sse transpose_note_attributes instead")`